src/k8s/api/v1/bootstrap_config.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// BootstrapConfig is used to seed cluster configuration when bootstrapping a new cluster.
+type BootstrapConfig struct {
+	// ClusterConfig
+	ClusterConfig UserFacingClusterConfig `json:"cluster-config,omitempty" yaml:"cluster-config,omitempty"`
+
+	// Seed configuration for the control plane (flat on purpose). Empty values are ignored
+	PodCIDR             *string `json:"pod-cidr,omitempty" yaml:"pod-cidr,omitempty"`
+	ServiceCIDR         *string `json:"service-cidr,omitempty" yaml:"service-cidr,omitempty"`
+	DisableRBAC         *bool   `json:"disable-rbac,omitempty" yaml:"disable-rbac,omitempty"`
+	SecurePort          *int    `json:"secure-port,omitempty" yaml:"secure-port,omitempty"`
+	CloudProvider       *string `json:"cloud-provider,omitempty" yaml:"cloud-provider,omitempty"`
+	K8sDqlitePort       *int    `json:"k8s-dqlite-port,omitempty" yaml:"k8s-dqlite-port,omitempty"`
+	DatastoreType       *string `json:"datastore-type,omitempty" yaml:"datastore-type,omitempty"`
+	DatastoreURL        *string `json:"datastore-url,omitempty" yaml:"datastore-url,omitempty"`
+	DatastoreCACert     *string `json:"datastore-ca-crt,omitempty" yaml:"datastore-ca-crt,omitempty"`
+	DatastoreClientCert *string `json:"datastore-client-crt,omitempty" yaml:"datastore-client-crt,omitempty"`
+	DatastoreClientKey  *string `json:"datastore-client-key,omitempty" yaml:"datastore-client-key,omitempty"`
```

Comment:
`DatastoreType` seems superfluous.  Help me know why its there?  

if we're using a socket for k8sdqlite, isn't that socket file just a DatastoreURL?

secondarily -- `DatastoreURL` should really be a `[]string` because i understand that the args are a comma joined set of URLs. right?

Reply:
We could use external datastore. If that's the case, we bootstrap the node with `datastore-type: external`

Reply:
@mateoflorido no i get that completely.  I just think that if you override the datastore url, you're very obviously using an "external" datastore.   from the kube-apiserver, it just wants to know the following:

```
etcd-servers
etcd-keyfile
etcd-certfile
etcd-cafile
```

it don't have a special name for the type.  This bootstrap config should represent only those things.  The "type" is a made up thing internally to this app but is unnecessary. 


the default should be `etcd-servers` = `DatastoreURL` = `unix:///var/snap..../k8sdqlite.socket`
and if someone wants it to be something else like a URL -- cool so be it.  Why is it necessary to drag this extra stringly typed field?

Reply:
@addyess The `DatastoreType` is external for etcd for example. We've got some logic based on the value of this type when we grab the cluster config from the bootstrap config. We're also using it to print our k8s status. In theory we could check if the DatastoreURL is set and that would mean we are dealing with an external datastore but that might make those switch statements quite ugly.